src/directive.js

```diff
@@ -4,15 +4,29 @@ const CONFIG_KEY = core.CONFIG_KEY
 export default {
   bind: (el, { value, modifiers }, vnode) => {
     el = core.getInputElement(el)
-    el.addEventListener('input', core.inputHandler, true)
-
     const config = core.normalizeConfig(value, modifiers)
     el[CONFIG_KEY] = { config }
 
     // set initial value
     core.updateValue(el, vnode, { force: config.prefill })
   },
 
+  inserted: (el) => {
+    el = core.getInputElement(el)
+    const config = el[CONFIG_KEY]
+    // prefer adding event listener to parent element to avoid Firefox bug which does not
+    // execute `useCapture: true` event handlers before non-capturing event handlers
+    const handlerOwner = el.parentElement || el
+
+    // use anonymous event handler to avoid inadvertently removing masking for all inputs within a container
+    const handler = (e) => core.inputHandler(e)
+
+    handlerOwner.addEventListener('input', handler, true)
+
+    config.handler = handler
+    config.cleanup = () => handlerOwner.removeEventListener('input', config.handler, true)
```

Comment:
I just realized we shouldn't need to store `handler` on the config object, since we can just capture the local variable in the lambda.

Reply:
Updated.